packages/react-hooks/src/hooks/steps.ts

```diff
@@ -0,0 +1,19 @@
+import {useState, useCallback} from 'react';
+
+export function useSteps(steps: number, initialStep = 0) {
+  const [currentStep, setStep] = useState(initialStep);
+
+  return {
```

Comment:
I would recommend returning an array here so that people can name the variable results conveniently with array destructuring, as `useState` does.

Reply:
You should be able to rename them as an object as well 

```tsx
const {
  currentStep: step,
  nextStep: next,
  previousStep: previous,
} = useSteps(5, 2);
```

Reply:
For sure, it's just more convenient as an array IMO but I don't feel strongly.